src/components/NavigatorSearch/useKeyboardNavigation.test.ts

```diff
@@ -0,0 +1,250 @@
+      resetFocus: mockResetFocus,
+      collectLinks: mockCollectLinks,
+      collectTableRows: mockCollectTableRows,
+      focusFirst: mockFocusFirst,
+      focusNext: mockFocusNext,
+      focusPrevious: mockFocusPrevious,
+      getActiveFocusedElement: mockGetActiveFocusedElement,
+      takeAction: mockTakeAction,
+    });
+
+    mockGetActiveFocusedElement.mockReturnValue(defaultMockElement);
+  });
+
+  describe("initialization", () => {
+    it("should return correct interface", () => {
+      const { result } = renderHook(() => useKeyboardNavigation({ query: "test" }));
+
+      expect(result.current).toMatchObject({
+        scope: expect.objectContaining({ current: null }),
+        resetFocus: mockResetFocus,
+        collectLinks: mockCollectLinks,
+        collectTableRows: mockCollectTableRows,
+      });
+    });
+
+    it("should register hotkeys with correct settings", () => {
+      renderHook(() => useKeyboardNavigation({ query: "test" }));
+
+      const expectedHotkeysSettings = {
+        enableOnFormTags: true,
+        scopes: ["command-menu"],
+      };
+
+      expect(useHotkeys).toHaveBeenCalledWith("tab", expect.any(Function), expectedHotkeysSettings);
+      expect(useHotkeys).toHaveBeenCalledWith("up", expect.any(Function), expectedHotkeysSettings);
+      expect(useHotkeys).toHaveBeenCalledWith(
+        "down",
+        expect.any(Function),
+        expectedHotkeysSettings,
+      );
+      expect(useHotkeys).toHaveBeenCalledWith(
+        "ctrl+enter, meta+enter",
+        expect.any(Function),
+        expectedHotkeysSettings,
+      );
+      expect(useHotkeys).toHaveBeenCalledWith(
+        "enter",
+        expect.any(Function),
+        expectedHotkeysSettings,
+      );
+    });
+  });
+
+  describe("query change effect", () => {
+    it("should focus first element and update aria active descendant when query changes", () => {
+      const { rerender } = renderHook(({ query }) => useKeyboardNavigation({ query }), {
+        initialProps: { query: "initial" },
+      });
+
+      expect(mockFocusFirst).toHaveBeenCalledTimes(1);
+      expect(mockGetActiveFocusedElement).toHaveBeenCalledTimes(1);
+      expect(mockUpdateAriaActiveDescendant).toHaveBeenCalledWith("test-element-id");
+
+      rerender({ query: "new query" });
+
+      expect(mockFocusFirst).toHaveBeenCalledTimes(2);
+      expect(mockGetActiveFocusedElement).toHaveBeenCalledTimes(2);
+      expect(mockUpdateAriaActiveDescendant).toHaveBeenCalledWith("test-element-id");
+    });
+
+    it("should not update aria active descendant when no active focused element", () => {
+      mockGetActiveFocusedElement.mockReturnValue(undefined);
+
+      renderHook(() => useKeyboardNavigation({ query: "test" }));
+
+      expect(mockFocusFirst).toHaveBeenCalled();
+      expect(mockGetActiveFocusedElement).toHaveBeenCalled();
+      expect(mockUpdateAriaActiveDescendant).not.toHaveBeenCalled();
+    });
```

Comment:
question: Shouldn't we set `aria-activedescendant` to empty string (or remove the property) when there is no active element? This should comply with WAI-ARIA spec: https://arc.net/l/quote/pqmvbkvk

> Sometimes states and properties are present in the DOM but have a zero-length string ("") as their value. Authors MAY specify a zero-length string ("") for any supported (but not required) state or property. User agents SHOULD treat state and property attributes with a value of "" the same as they treat an absent attribute.

issue: https://github.com/w3c/aria/issues/501